### PR TITLE
[BUG] Prevent infinite loop in Data API recursive call

### PR DESCRIPTION
**Pull request by gonzalozawa**

### CODE REVIEW

File: LearnositySDK/Request/DataApi.cs [hunk 1]
```diff
@@ -160,7 +160,7 @@ private JsonObject handleRequestRecursive(string url, JsonObject securityPacket,
                     }
                 }
             }
-            while(Tools.array_key_exists("next", meta));
+            while(Tools.array_key_exists("next", meta) && !Tools.empty(data.getJsonObject("data")));
```
Comment (karoltarasiuk) [inline LearnositySDK/Request/DataApi.cs]: @gonzalozawa hey, makes sense, I'd only suggest to do a small refactor to make it better aligned with PHP SDK: https://github.com/Learnosity/learnosity-sdk-php/blob/master/src/LearnositySdk/Request/DataApi.php

Comment (gonzalozawa) [inline LearnositySDK/Request/DataApi.cs]: Hi @karoltarasiuk, sorry for the delay in getting back to you. When you mentioned a refactor to align with PHP SDK. Are you sure about that? Or should we refactor PHP SDK and remove the if from line 79 (https://github.com/Learnosity/learnosity-sdk-php/blob/master/src/LearnositySdk/Request/DataApi.php#L79) and add that condition to the while? Let me know your thoughts. Cheers!

Comment (karoltarasiuk) [inline LearnositySDK/Request/DataApi.cs]: @gonzalozawa we've fixed infinite loop in PHP SDK here: https://github.com/Learnosity/learnosity-sdk-php/commit/7c4092eecb616eb658907e4e96561721abf0073e

It's similar to your fix but uses `$requestPacket` as a condition for `while` loop. I suggested to have the same solution in .NET SDK, as it's easier then to compare both SDKs. .NET was written based on PHP one and let's keep it this way.

**Comment by gonzalozawa** [comment]
Thanks @karoltarasiuk for your comments. I have applied the changes to match the PHP SDK. Could you please have a look and let me know your thoughts? Cheers!

**Comment by karoltarasiuk** [comment]
@gonzalozawa code looks good now. Didn't do functional though, as it's quite a hassle here to do it, and we're close to wrapping up a sprint. If you want me to do the functional it will have to wait and I will need JIRA ticket. If you're happy with it though, you can ho ahead and merge it.

**Pull request merged**